test/run

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2016 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+import sys
+
+
+# FIXME: Learn image-prepare to read image from TEST_OS
+def prepare(prefix, value, image):
+    sys.stderr.write("Preparing image: building and installing Cockpit ...\n")
+
+    try:
+        # Download all the additional images so that even older branches find them
+        subprocess.check_call([os.path.join("bots", "image-download"),
+                               "candlepin", "fedora-stock", "fedora-23-stock",
+                               "ipa", "openshift", "selenium"])
+
+    except subprocess.CalledProcessError:
+        return "Downloading of additional images failed"
+
+    # COMPAT: Create a legacy tmp/run directory to prevent races during testing
+    os.makedirs(os.path.join("test", "tmp", "run"), exist_ok=True)
+
+    # Now actually run the prepare tooling
+    cmd = [os.path.join("test", "image-prepare")]
+    if prefix in ["image", "container"]:
+        cmd += ["--containers"]
+    cmd += ["--verbose"]
+
+    # Flush our own output before invoking the command
+    sys.stdout.flush()
+    sys.stderr.flush()
+
+    try:
+        # Do the basic prepare
+        subprocess.check_call(cmd + [image])
+
+        # For containers tests we install to openshift too
+        if value == "kubernetes":
+            subprocess.check_call(cmd + ["--install-only", "openshift"])
+
+    except subprocess.CalledProcessError:
+        return "Preparation of testable image failed"
+    return ""
+
+
+def main():
+    scenario = os.environ.get("TEST_SCENARIO", "verify")
+    jobs = os.environ.get("TEST_JOBS", 1)
+    image = os.environ.get("TEST_OS")
+
+    (prefix, _, value) = scenario.partition("-")
+    ret = prepare(prefix, value, image)
+
+    # Figure out what to do next
+    if prefix == "verify":
+        cmd = ["timeout", "120m", os.path.join("test", "verify", "run-tests"),
```

Comment:
As discussed yesterday, let's please drop the timeouts here. The bots have a global timeout in tests-invoke, and each test has its own timeout, too. It was rather confusing last time having to change them in three different places.

Reply:
All of them, or just this 120m? some of them have shorter timeout, not that it makes that much of a difference.

Reply:
Yes, all of them. test/run can become quite simple then (this is why I thought a shell script would suffice, as it's really not much more than a `case` switch and calling a few programs -- but py is fine!).